protobuf_serialization/reader.nim

```diff
@@ -137,6 +184,9 @@ proc readValueInternal[T: object](stream: InputStream, value: var T, silent: boo
             stream.readFieldPackedInto(fieldVar, header, ProtoType)
           else:
             stream.readFieldInto(fieldVar, header, ProtoType)
+        elif ProtoType is ref and defined(ConformanceTest):
```

Comment:
why does the conformance test need this? there are no ref semantics in protobuf afair

Reply:
Can't work:
```nim
import stew/results
type
  A = object
    b: Opt[B]

  B = object
    a: Opt[A]
```

```
test.nim(3, 3) Error: illegal recursion in type 'A'
```